scripts/checks/verify_supply_chain.py

```
@@ -166,16 +167,146 @@ def verify_workflow_coverage() -> list[str]:
+    workflow_paths = sorted(Path(".github/workflows").glob("*.yml")) + sorted(
+        Path(".github/workflows").glob("*.yaml")
+    )
+    root_working_directories = {"", ".", "./", "${{ github.workspace }}"}
+
+    for path in workflow_paths:
+        step_working_directory = ""
+        step_uses_workspace_exec = False
+
+        def record_step_violation() -> None:
+            if (
+                step_uses_workspace_exec
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `step_uses_workspace_exec`

(B023)